File: www/fullcgi/cgi-bin/fullcgi.py
import html
import os

print("Content-Type: text/html")
print()

print("<!DOCTYPE html>")
print('<html lang="en">')
print("<head><title>Environment Variables</title></head>")
print("<body>")
print("<table border='1'>")
print("<tr><th>Variable</th><th>Value</th></tr>")

for key, value in os.environ.items():
    print(f"<tr><td>{html.escape(key)}</td><td>{html.escape(value)}</td></tr>")

print("</table>")
print("</body>")
print("</html>")

File: www/fullcgi/cgi-bin/fullcgi.php
<?php
header("Content-Type: text/html");

echo "<!DOCTYPE html>";
echo '<html lang="en">';
echo "<head><title>Environment Variables</title></head>";
echo "<body>";
echo "<table border='1'>";
echo "<tr><th>Variable</th><th>Value</th></tr>";

foreach ($_SERVER as $key => $value) {
    echo "<tr><td>" . htmlspecialchars($key) . "</td><td>" . htmlspecialchars($value) . "</td></tr>";
}

echo "</table>";
echo "</body>";
echo "</html>";
?>
